sgt-back-end/index.js

```diff
@@ -0,0 +1,165 @@
+               values ($1, $2, $3)
+               returning *
+               `;
+  const params = [req.body.name, req.body.course, req.body.score];
+  if (params[0] === undefined || params[1] === undefined || params[2] === undefined) {
+    res.status(400).json({ error: 'all fields are required' });
+    return;
+  } else if (req.body.score < 0 || req.body.score > 100 || !Number.isInteger(req.body.score)) {
+    res.status(400).json({ error: 'score is required to be a number between 0 and 100' });
+    return;
+  }
+  db.query(sql, params)
+    .then((result) => {
+      const data = result.rows[0];
+      res.status(201).json(data);
+    })
+    .catch((err) => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({ error: '"gradeId" must be a positive integer' });
+    return;
+  } else if (req.body.name === undefined || req.body.course === undefined || req.body.score === undefined) {
+    res.status(400).json({ error: 'all fields are required' });
+  }
+  const sql = `update "grades"
+                set "name" = $2,
+                    "course" = $3,
+                    "score" = $4
+                where "gradeId" = $1
+                returning *
+                `;
+  const params = [gradeId, req.body.name, req.body.course, req.body.score];
+  db.query(sql, params)
+    .then((result) => {
+      const data = result.rows[0];
+      if (!data) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+        return;
+      }
+      res.json(data);
+    })
+    .catch((err) => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.delete('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({ error: '"gradeId" must be a positive integer' });
+    return;
+  }
+  const sql = `delete from "grades"
+                where "gradeId" = $1
+                returning *
+                `;
+  const params = [gradeId];
+  db.query(sql, params)
+    .then((result) => {
+      const data = result.rows[0];
+      if (!data) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+        return;
+      }
+      res.status(204).json(data);
+    })
+    .catch((err) => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.listen(3000, (err) => {
+  if (err) {
+    console.error('Something went wrong', err);
+  }
+  // eslint-disable-next-line no-console
+  console.log('Listening on port 3000');
```

Comment:
If you get an error, you will still log the "Listening" message, which is probably not what you want.